Homogeneous_kernel/include/CGAL/Homogeneous/function_objects.h

```diff
@@ -4709,6 +4710,33 @@ namespace HomogeneousKernelFunctors {
         ? ON_ORIENTED_BOUNDARY
         : -ot;
     }
+
+    result_type
+    operator()(const Segment_2& s, const Triangle_2& t) const
+    {
+      typename K::Construct_source_2 source;
+      typename K::Construct_target_2 target;
+      typename K::Construct_circumcenter_2 circumcenter;
+      typename K::Orientation_2 orientation;
+
+      const Point_2 a = source(s);
+      const Point_2 b = target(s);
+      const Point_2 cc = circumcenter(t);
+
+      CGAL::Orientation o_abc = orientation(a, b, cc);
```

Comment:
I'm not in favor of handling degenerate cases. There should be assertions that input are not degenerate (also for cartesian btw)

Reply:
I added code to deal with degenerate inputs. This test only deals with a degenerate configuration where `cc` is collinear with `ss`